internal/backend/svc/svc.go

```diff
@@ -186,6 +187,20 @@ func makeFxOpts(cfg *Config, opts RunOptions) []fx.Option {
 		Component("runtimes", configset.Empty, fx.Provide(runtimes.New)),
 
 		Component("healthcheck", configset.Empty, fx.Provide(healthchecker.New)),
+		Component(
+			"schedule",
+			configset.Empty,
+			fx.Provide(schedule.New),
+			// fx.Provide(func(sch schedule.Scheduler) sdkservices.Scheduler { return sch }),
+			// fx.Invoke(func(lc fx.Lifecycle, sch schedule.Scheduler) { HookOnStart(lc, sch.Start) }),
+		),
+		Component(
```

Comment:
why is this a separate component? why is it not included inside schedule?

Reply:
due to import loop. triggers need scheduler and it needs triggers. I splitted this to two components. Scheduler which is responsible for starting/updating/deleting temporal schedules and scheduleWorkflow

Reply:
add TODO to consider this

Reply:
TODO: consider design where the workflow is an implementation detail of the scheduler and does not need to be exposed.

Reply:
added todo.